Ginger/GingerCoreNET/Run/Context.cs

```diff
@@ -112,6 +131,144 @@ public TargetBase Target
             }
         }
 
+        private string mTargetApplication;
+        public string TargetApplication
+        {
+            get
+            {
+                return mTargetApplication;
+            }
+            set
+            {
+                if (mTargetApplication != value)
+                {
+                    mTargetApplication = value;
+                    OnPropertyChanged(nameof(TargetApplication));
+                }
+            }
+        }
+
+        private ApplicationAgent mAppAgent;
+        public ApplicationAgent AppAgent
```

Comment:
no need AppAgent we already have Target (App) and Agent sperately

Reply:
Removed